ams/metric-reporters/prometheus-reporter/src/main/java/com.netease.arctic.metrics.reporter.promethues/MetricCollector.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.metrics.reporter.promethues;
+
+import com.netease.arctic.ams.api.metrics.Counter;
+import com.netease.arctic.ams.api.metrics.Gauge;
+import com.netease.arctic.ams.api.metrics.Metric;
+import com.netease.arctic.ams.api.metrics.MetricDefine;
+import com.netease.arctic.ams.api.metrics.MetricKey;
+import com.netease.arctic.ams.api.metrics.MetricSet;
+import com.netease.arctic.ams.api.metrics.MetricType;
+import io.prometheus.client.Collector;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/** Metric type converter for prometheus api */
+public class MetricCollector extends Collector {
+  MetricSet metrics;
+
+  public MetricCollector(MetricSet metrics) {
+    this.metrics = metrics;
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+    Map<MetricKey, Metric> registeredMetrics = metrics.getMetrics();
+
+    Map<MetricDefine, List<MetricKey>> metricDefineMap =
+        registeredMetrics.keySet().stream()
+            .collect(
+                Collectors.groupingBy(
+                    MetricKey::getDefine,
+                    Collectors.mapping(Function.identity(), Collectors.toList())));
+    return metricDefineMap.entrySet().stream()
+        .map(entry -> createFamilySample(entry.getKey(), entry.getValue(), registeredMetrics))
+        .collect(Collectors.toList());
+  }
+
+  private MetricFamilySamples createFamilySample(
+      MetricDefine define, List<MetricKey> keys, Map<MetricKey, Metric> registeredMetrics) {
+
+    List<MetricFamilySamples.Sample> samples = Lists.newArrayList();
+    for (MetricKey key : keys) {
+      Metric metric = registeredMetrics.get(key);
+
+      MetricFamilySamples.Sample sample =
+          new MetricFamilySamples.Sample(
```

Comment:
There may be two points that can be improved.

1. According to the official naming [guidelines](https://prometheus.io/docs/practices/naming/) of prometheus and the practice of some excellent open source projects, you can add the relevant service name before the reported indicator name, such as the prefix of `amoro_`. Avoid conflict with indicator names of other services.

2. Also according to prometheus [guidelines](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels), The characters of the name and label should be checked before reporting whether they comply with the rules `[a-zA-Z_:][a-zA-Z0-9_:]*`, otherwise they will not be reported to prometheus. One of the checking processing methods It is to uniformly convert characters that do not comply with the rules into `_`

Reply:
done.